src/field/arena.ts

```diff
@@ -122,10 +118,11 @@ export class Arena {
     const timeOfDay = this.getTimeOfDay();
     if (timeOfDay !== this.lastTimeOfDay) {
       this.pokemonPool = {};
-      for (const tier of Object.keys(biomePokemonPools[this.biomeType])) {
-        this.pokemonPool[tier] = Object.assign([], biomePokemonPools[this.biomeType][tier][TimeOfDay.ALL]).concat(
-          biomePokemonPools[this.biomeType][tier][timeOfDay],
-        );
+      for (const tier of Object.keys(allBiomes.get(this.biomeType).pokemonPool)) {
+        this.pokemonPool[tier] = Object.assign(
+          [],
+          allBiomes.get(this.biomeType).pokemonPool[tier][TimeOfDay.ALL],
+        ).concat(allBiomes.get(this.biomeType).pokemonPool[tier][timeOfDay]);
```

Comment:
NIT: There is also a `Object.entries()` method which returns an array of `[key, value]` so you don't have to use `allBiomes.get(this.biomeType).pokemonPool[tier]` inside the loop

Reply:
`Object.assign` can also be often replaced with the `split` operator `...`

```suggestion
      for (const [tier, pool] of Object.entries(allBiomes.get(this.biomeType).pokemonPool)) {
        this.pokemonPool[tier] = [
          ...pool[TimeOfDay.ALL],
          ...pool[timeOfDay],
        ];
      }
```